command/agentproxyshared/cache/lease_cache.go

```diff
@@ -585,7 +588,7 @@ func (c *LeaseCache) cacheStaticSecret(ctx context.Context, req *SendRequest, re
 	// The index already exists, so all we need to do is add our token
 	// to the index's allowed token list, then re-store it
 	if indexFromCache != nil {
-		indexFromCache.Tokens = append(indexFromCache.Tokens, req.Token)
+		indexFromCache.Tokens[req.Token] = struct{}{}
```

Comment:
hmm, yeah, you're right, good catch! There's no reason to lock the `index` here since it's not possible for it to be accessed by another thread (it's not in the db yet)